src/interceptors/ClientRequest/utils/normalizeHttpRequestParams.ts

```diff
@@ -98,17 +102,27 @@ export function normalizeHttpRequestParams(
     const resolvedUrl = new URL(legacyUrl.href)
 
     return args[1] === undefined
-      ? normalizeHttpRequestParams(resolvedUrl)
+      ? normalizeHttpRequestParams(defaultProtocol, resolvedUrl)
       : typeof args[1] === 'function'
-      ? normalizeHttpRequestParams(resolvedUrl, args[1])
-      : normalizeHttpRequestParams(resolvedUrl, args[1], args[2])
+      ? normalizeHttpRequestParams(defaultProtocol, resolvedUrl, args[1])
+      : normalizeHttpRequestParams(
+          defaultProtocol,
+          resolvedUrl,
+          args[1],
+          args[2]
+        )
   }
   // Handle a given RequestOptions object as-is
   // and derive the URL instance from it.
   else if (isObject(args[0])) {
     options = args[0]
     debug('given request options:', options)
 
+    // When handling a `RequestOptions` object without an explicit "protocol",
+    // infer the protocol from the request issuing module (http/https).
+    options.protocol = options.protocol || defaultProtocol
```

Comment:
Protocol coercion is applicable only when passing `RequestOptions` object as the argument to `http.ClientRequest`. In that case, fallback to the default protocol inferred from the request module if no explicit protocol was provided.

> Some libraries (i.e. `superagent`) do not provide any protocol explicitly. They do issue a request using the right request module, however (http/https).